Tests/Reader/Makefile

```diff
@@ -0,0 +1,6 @@
+ALL: compile run
+compile: reader.cpp main.cpp
+	g++ -o compile reader.cpp main.cpp -I.
```

Comment:
This version of the makefile doesn't use patterns. You could automatically generate rules to create object files like this:
```
%.o: %.cpp reader.hpp
	g++ -o $@ $^ -I.
```
The `%.o: %.cpp` creates a rule to generate an object file for each `cpp` file that depends on `header.hpp`. In our case it resolves to
- `reader.o: reader.cpp reader.hpp`
- `main.o: main.cpp reader.hpp`

Then the script line in the tab to generate the target uses automatic variables. The `$@` means the target. So in this case the `.o` file. The `$^` means all dependencies. So the `.cpp` file and the `reader.hpp` file.

Then you will need a linker line such as

```
reader: reader.o main.o
	g++ -o $@ $^
```